File: src/main/java/com/dh/ondot/member/domain/enums/MapProvider.java
package com.dh.ondot.member.domain.enums;

import com.dh.ondot.core.exception.UnsupportedException;

import java.util.Locale;

import static com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_MAP_PROVIDER;

public enum MapProvider {
    KAKAO,
    NAVER;

    public static MapProvider from(String ringTone) {
        try {
            return MapProvider.valueOf(ringTone.toUpperCase(Locale.ENGLISH));
        } catch (IllegalArgumentException e) {
            throw new UnsupportedException(UNSUPPORTED_MAP_PROVIDER, ringTone);
        }
    }
}

File: src/main/kotlin/com/dh/ondot/member/domain/enums/MapProvider.kt
package com.dh.ondot.member.domain.enums

import com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_MAP_PROVIDER
import com.dh.ondot.core.exception.UnsupportedException

enum class MapProvider {
    KAKAO,
    NAVER,
    APPLE,
    ;

    companion object {
        fun from(mapProvider: String): MapProvider =
            entries.find { it.name.equals(mapProvider, ignoreCase = true) }
                ?: throw UnsupportedException(UNSUPPORTED_MAP_PROVIDER, mapProvider)
    }
}
